src/Mapper/Source/JsonSource.php

```diff
@@ -29,10 +32,10 @@ final class JsonSource implements IteratorAggregate
      */
     public function __construct(string $jsonSource)
     {
-        $source = json_decode($jsonSource, true);
-
-        if ($source === null) {
-            throw new InvalidJson($jsonSource);
+        try {
+            $source = json_decode($jsonSource, true, flags: JSON_THROW_ON_ERROR);
```

Comment:
```suggestion
            $source = json_decode($jsonSource, associative: true, flags: JSON_THROW_ON_ERROR);
```

While we're there, we should ease the reading for `associative` as well. 🙂

Reply:
Okay, done. Let me know if you prefer splitting the function call across multiple lines (and if "yes" whether or not to use a trailing comma). Also let me know if I should squash my commits.

Reply:
That's very good like it is, thanks for asking. I'll do the squashing myself!